### PR TITLE
DOCSP-45010: Replica Set Operations

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-45010>

### Staging Links

<li><a href=https://deploy-preview-322--mongodb-docs-csharp.netlify.app/fundamentals/read-write-configuration>fundamentals/read-write-configuration</a></li><li><a href=https://deploy-preview-322--mongodb-docs-csharp.netlify.app/fundamentals>fundamentals</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals.txt [hunk 1]
```diff
@@ -30,6 +30,7 @@ Fundamentals
    Time Series Collections </fundamentals/time-series>
    In-Use Encryption </fundamentals/encrypt-fields>
    Search Geospatially </fundamentals/geo>
+   Replica Set Operations </fundamentals/read-write-configuration>
```
Comment (stephmarie17) [inline source/fundamentals.txt:33]: Should this display on this page in the [deploy preview](https://deploy-preview-322--mongodb-docs-csharp.netlify.app/fundamentals/) at the end of the list? I don't see it for some reason 🤔

File: source/fundamentals/read-write-configuration.txt [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+The ``ReadConcern`` option and ``WithReadConcern()`` method both accept a single parameter
+that specifies the read concern level.
+
+You can set the following read concern levels:
+
+- ``ReadConcern.Local``: The query returns the instance's most recent data. Provides no guarantee
+  that the data has been written to a majority of the replica set members.
+- ``ReadConern.Available``: The query returns the instance's most recent data.
+  Provides no guarantee that the data has been written to a majority of the
+  replica set members. ``ReadConcern.AVAILABLE`` is not available for use with
```
Comment (stephmarie17) [inline source/fundamentals/read-write-configuration.txt:117]: [q] Is this meant to be capitalized:
```suggestion
  replica set members. ``ReadConcern.Available`` is not available for use with
```

File: source/fundamentals/read-write-configuration.txt [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/ReplicaSetConfigs.cs
+   :start-after: start-read-concern-collection
+   :end-before: end-read-concern-collection
+   :language: csharp
+   :dedent:
+
+To learn more about read concern, see :manual:`Read Concern
+<reference/read-concern>` in the {+mdb-server+} manual.
```
Comment (stephmarie17) [inline source/fundamentals/read-write-configuration.txt:151]: This link was a 404 I think it's missing a slash:
```suggestion
<reference/read-concern/>` in the {+mdb-server+} manual.
```

File: source/fundamentals/read-write-configuration.txt [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+Write concern specifies the level of acknowledgement requested from MongoDB for
+write operations before the operation successfully returns. Operations that
+don't specify an explicit write concern inherit the global default write concern
+setting.
+
+You can set the write concern by setting the ``WriteConcern`` option on a
+``MongoClientSettings``, ``MongoDatabaseSettings``, or ``MongoCollectionSettings`` object,
+or by using the ``WithWriteConcern()`` method of a client, database, or collection instance.
+
+The ``WriteConcern`` option and ``WithWriteConcern()`` method both accept a
```
Comment (stephmarie17) [inline source/fundamentals/read-write-configuration.txt:51]: [s] Omit additional word, I think it still makes sense without it:
```suggestion
The ``WriteConcern`` option and ``WithWriteConcern()`` method accept a
```

**Review by stephmarie17 — APPROVED** [review]
Nice work 🎇 LGTM with a few suggestions and a URL fix.

File: source/fundamentals.txt (hunk 1 above)
Comment (mcmorisi) [reply source/fundamentals.txt:33]: Forgot the link! Will add in revision.

File: source/fundamentals/read-write-configuration.txt [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+.. literalinclude:: /includes/fundamentals/code-examples/ReplicaSetConfigs.cs
+   :start-after: start-write-concern-collection
+   :end-before: end-write-concern-collection
+   :language: csharp
+   :dedent:
+
+.. note:: Collections and Databases are Immutable
+
+   ``IMongoDatabase`` and ``IMongoCollection`` instances are immutable. When you
```
Comment (adelinowona) [inline source/fundamentals/read-write-configuration.txt:87]: `IMongoClient` instances are also immutable as you can't modify settings after creating them. If you need to apply different settings to an `IMongoClient`, `IMongoDatabase`, or `IMongoCollection`, you must use one of the `With*` methods. These methods are designed to always return new instances with the specified settings. I recommend updating this note to reflect this information.

**Review by adelinowona — APPROVED** [review]
LGTM

**Pull request merged**